### PR TITLE
[JENKINS-26462] Make plugin compatible with maven builder extraction from the core

### PR DESCRIPTION
**Pull request by alvarolobato**
[JENKINS-26462](https://issues.jenkins-ci.org/browse/JENKINS-26462)
This is a downstream PR of [PR-2352](https://github.com/jenkinsci/jenkins/pull/2352)

As a consequence os the extraction from the core of the maven builder and the creation of the Maven Builder Plugin, these changes need to be made in order to make the plugin compatible with the new versions of the core.

@reviewbybees esp @jglick @oleg-nenashev

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -43,8 +43,8 @@ THE SOFTWARE.
   <url>http://wiki.jenkins-ci.org/display/JENKINS/Maven+Project+Plugin</url>
 
   <properties>
-    <jenkins.version>1.580.1</jenkins.version>
-    <java.level>6</java.level>
+    <jenkins.version>2.4-SNAPSHOT</jenkins.version>
```
Comment (jglick) [inline pom.xml]: You need to make your upstream PR up to date with the `master` branch.

File: pom.xml [hunk 2]
```diff
@@ -43,8 +43,8 @@ THE SOFTWARE.
...
+    <java.level>7</java.level>
```
Comment (jglick) [inline pom.xml]: Just delete, this is the default.

File: pom.xml (hunk 1 above)
Comment (alvarolobato) [inline pom.xml]: Changed.

File: pom.xml (hunk 2 above)
Comment (alvarolobato) [inline pom.xml]: Changed

File: src/main/java/hudson/maven/MavenPluginMavenSelectorImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * {@link MavenSelector} extension for {@link MavenModule} and {@link MavenModuleSet} {@link Item}s
+ */
+@Extension
+public class MavenPluginMavenSelectorImpl extends MavenSelector {
+
+    @Override
+    public boolean isApplicable(Class<? extends Item> itemClass) {
+        if (itemClass.isAssignableFrom(MavenModule.class) || itemClass.isAssignableFrom(MavenModuleSet.class)) {
```
Comment (olivergondza) [inline src/main/java/hudson/maven/MavenPluginMavenSelectorImpl.java]: `return itemClass.isAssignableFrom(MavenModule.class) || itemClass.isAssignableFrom(MavenModuleSet.class)`.

Comment (alvarolobato) [inline src/main/java/hudson/maven/MavenPluginMavenSelectorImpl.java]: duhhh, what was I thinking on... changing it.

**Comment by jglick** [comment]
Could not resolve dependencies for project org.jenkins-ci.main:maven-plugin:hpi:2.14-SNAPSHOT: The following artifacts could not be resolved: org.jenkins-ci.plugins:maven-builder-plugin:jar:1.0-SNAPSHOT, org.jenkins-ci.main:jenkins-core:jar:2.13-SNAPSHOT, org.jenkins-ci.main:jenkins-war:jar:war-for-test:2.13-SNAPSHOT, org.jenkins-ci.main:jenkins-war:war:2.13-SNAPSHOT: Could not find artifact org.jenkins-ci.plugins:maven-builder-plugin:jar:1.0-SNAPSHOT in repo.jenkins-ci.org (http://repo.jenkins-ci.org/public/)

means you need to `deploy` snapshot dependencies of upstream PRs.

**Comment by aheritier** [comment]
I'm sorry @alvarolobato but I tried to work on this plugin recently thus now it has some conflicts. If required I can help you to rebase it on top of master

**Comment by olamy** [comment]
old PR and probably not needed anymore so closing it.

**Comment by jglick** [comment]
I think we should still do this, but the PRs can be recreated whenever that happens.

**Pull request closed without merging**